File: road-runner-quickstart-master/TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robot/actions/IntakeClawClose.java
package org.firstinspires.ftc.teamcode.robot.actions;

import androidx.annotation.NonNull;

import com.acmerobotics.dashboard.telemetry.TelemetryPacket;
import com.acmerobotics.roadrunner.Action;

import org.firstinspires.ftc.teamcode.robot.Intake;

public class IntakeClawClose implements Action {
    private boolean init = false;
    private Intake i;

    @Override
    public boolean run(@NonNull TelemetryPacket packet) {
        if(!init) {
            i.clawOpen();
            init = true;
        }

        return false;
    }

    public IntakeClawClose(Intake li) {
        i = li;
    }

    public Action intakeClawClose(Intake li) {
        return new IntakeClawClose(li);
    }
}

File: road-runner-quickstart-master/TeamCode/src/main/java/org/firstinspires/ftc/teamcode/dairy/actions/IntakeClawClose.kt
package org.firstinspires.ftc.teamcode.dairy.actions

import com.acmerobotics.dashboard.telemetry.TelemetryPacket
import com.acmerobotics.roadrunner.Action
import org.firstinspires.ftc.teamcode.dairy.subsystems.IntakeClaw
import org.firstinspires.ftc.teamcode.dairy.util.Waiter

class IntakeClawClose() : Action {
    private var init = false
    private lateinit var waiter:Waiter

    override fun run(p: TelemetryPacket): Boolean {
        if (!init) {
            waiter = Waiter()
            waiter.start(200)
            IntakeClaw.claw_pos = IntakeClaw.claw_close
            init = true
        }

        return waiter.isDone
    }

    companion object {
        fun intakeClawClose(): Action {
            return IntakeClawClose()
        }
    }
}